pkg/signer/request-signature-streaming.go

```diff
@@ -294,6 +304,7 @@ func StreamingSignV4(req *http.Request, accessKeyID, secretAccessKey, sessionTok
 		chunkNum:        1,
 		totalChunks:     int((dataLen+payloadChunkSize-1)/payloadChunkSize) + 1,
 		lastChunkSize:   int(dataLen % payloadChunkSize),
+		sh256:           sh256,
```

Comment:
Maybe let's define a default sha256 if sh256 is nil ? this is a public package

Reply:
We always set a fallback if unset: https://github.com/minio/minio-go/blob/efc53fd9e0f1d982354d3168aa437165b19fd110/api.go#L253